presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -322,8 +325,19 @@ private static boolean checkSchemaMatch(org.apache.parquet.schema.Type parquetTy
             switch (prestoType) {
                 case ROW:
                     if (groupType.getFields().size() == type.getTypeParameters().size()) {
+                        checkState(type instanceof RowType, "It must be a RowType here.");
+                        RowType rowType = (RowType) type;
+                        Map<String, Type> prestoFieldMap = rowType.getFields().stream().collect(
+                                Collectors.toMap(
+                                        f -> f.getName().get(),
+                                        f -> f.getType()));
                         for (int i = 0; i < groupType.getFields().size(); i++) {
-                            if (!checkSchemaMatch(groupType.getFields().get(i), type.getTypeParameters().get(i))) {
+                            org.apache.parquet.schema.Type parquetFieldType = groupType.getFields().get(i);
+                            Type prestoFieldType = prestoFieldMap.get(parquetFieldType.getName());
+                            if (prestoFieldType == null) {
+                                prestoFieldType = prestoFieldMap.get(parquetFieldType.getName() + "_");
```

Comment:
why use parquetFieldType.getName() + "_"?
if prestoFieldType == null, we should return false?

Reply:
Good call, this feature is twitter specific, which shouldn't in this pr